gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemNewIntegrationTest.java

```diff
@@ -101,11 +120,62 @@ public void mkdirs_shouldCreateNewDirectory() throws Exception {
     String dirObject = getTestResource();
     URI dirObjectUri = new URI("gs://" + bucketName).resolve(dirObject);
 
+    gcsfsIHelper.mkdir(
+        bucketName, dirObject); // create directory before hand without tracking requests
     gcsFs.mkdir(dirObjectUri);
 
     assertThat(gcsRequestsTracker.getAllRequestStrings())
         .containsExactly(
-            uploadRequestString(bucketName, dirObject + "/", /* generationId= */ null));
+            uploadRequestString(bucketName, dirObject + "/", /* generationId= */ 1),
```

Comment:
Probably we would want to check that generation ID was `0` in this case. Maybe by adding `getAllRawRequestStrings` and adding overloaded `uploadRequestString` method with `replaceGenerationId` parameter, like here: https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/7891c3d898944c7f47b70fe4e7d276bc4da5c3c1/gcsio/src/test/java/com/google/cloud/hadoop/gcsio/TrackingHttpRequestInitializer.java#L259

Reply:
Done